File: app/src/main/java/com/o4x/musical/ui/activities/PlayerActivity.java
package com.o4x.musical.ui.activities;

import android.animation.ValueAnimator;
import android.graphics.Color;
import android.os.Bundle;

import androidx.annotation.ColorInt;

import com.o4x.musical.R;
import com.o4x.musical.ui.activities.base.AbsMusicServiceActivity;
import com.o4x.musical.ui.fragments.player.AbsPlayerFragment;
import com.o4x.musical.ui.fragments.player.flat.FlatPlayerFragment;

import code.name.monkey.appthemehelper.util.ColorUtil;

public class PlayerActivity extends AbsMusicServiceActivity implements FlatPlayerFragment.Callbacks {

    private int navigationBarColor;
    private int taskColor;
    private boolean lightStatusBar;

    private AbsPlayerFragment playerFragment;

    private ValueAnimator navigationBarColorAnimator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player);

        setDrawUnderBar();

        playerFragment = new FlatPlayerFragment();

        getSupportFragmentManager().beginTransaction().replace(R.id.player_fragment_container, playerFragment).commit();
        getSupportFragmentManager().executePendingTransactions();
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (navigationBarColorAnimator != null) navigationBarColorAnimator.cancel(); // just in case
    }

    @Override
    protected void onStart() {
        super.onStart();
        // setting fragments values
        int playerFragmentColor = playerFragment.getPaletteColor();
        this.setLightStatusBar(false);
        super.setTaskDescriptionColor(playerFragmentColor);
        super.setNavigationBarColor(Color.TRANSPARENT);

        playerFragment.setMenuVisibility(true);
        playerFragment.setUserVisibleHint(true);
        playerFragment.onShow();
    }

    @Override
    protected void onStop() {
        super.onStop();
        // restore values
        super.setLightStatusBar(lightStatusBar);
        super.setTaskDescriptionColor(taskColor);
        super.setNavigationBarColor(navigationBarColor);

        playerFragment.setMenuVisibility(false);
        playerFragment.setUserVisibleHint(false);
        playerFragment.onHide();
    }

    @Override
    public void onPaletteColorChanged() {
        int playerFragmentColor = playerFragment.getPaletteColor();
        super.setTaskDescriptionColor(playerFragmentColor);
        super.setLightNavigationBar(ColorUtil.INSTANCE.isColorLight(playerFragmentColor));
    }

    @Override
    public void setNavigationBarColor(int color) {
        this.navigationBarColor = color;
        if (navigationBarColorAnimator != null) navigationBarColorAnimator.cancel();
        super.setNavigationBarColor(color);
    }

    @Override
    public void setLightStatusBar(boolean enabled) {
        lightStatusBar = enabled;
        super.setLightStatusBar(enabled);
    }

    @Override
    public void setTaskDescriptionColor(@ColorInt int color) {
        this.taskColor = color;
        super.setTaskDescriptionColor(color);
    }
}

File: app/src/main/java/com/o4x/musical/ui/activities/PlayerActivity.kt
package com.o4x.musical.ui.activities

import android.graphics.Color
import android.os.Bundle
import android.view.MenuItem
import androidx.annotation.ColorInt
import androidx.appcompat.app.AppCompatDelegate
import androidx.appcompat.app.AppCompatDelegate.setDefaultNightMode
import com.o4x.appthemehelper.util.ColorUtil.isColorLight
import com.o4x.musical.R
import com.o4x.musical.prefs.PreferenceUtil
import com.o4x.musical.ui.activities.base.AbsMusicServiceActivity
import com.o4x.musical.ui.fragments.player.PlayerFragment

class PlayerActivity : AbsMusicServiceActivity() {

    lateinit var playerFragment: PlayerFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_player)
        setDrawUnderBar()

        playerFragment = PlayerFragment()
        supportFragmentManager.beginTransaction()
            .replace(R.id.player_fragment_container, playerFragment).commit()
        supportFragmentManager.executePendingTransactions()
        setLightStatusBar(false)
        setNavigationBarColor(Color.TRANSPARENT)
    }

    override fun setTheme() {
        setTheme(R.style.Theme_Musical_Black)
    }

    override fun onStart() {
        super.onStart()
        // setting fragments values
        playerFragment.setMenuVisibility(true)
        playerFragment.userVisibleHint = true
    }

    override fun onStop() {
        super.onStop()
        playerFragment.setMenuVisibility(false)
        playerFragment.userVisibleHint = false
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}